starknet/src/server/stream.rs

```diff
@@ -74,6 +79,31 @@ where
 
         ResponseStream::new(data_stream).instrument(stream_span)
     }
+
+    async fn sync_status(self, rpc_url: Url) -> StatusResponse {
```

Comment:
I believe the url/provider should be a property of the server. 

The best way is to reuse the `Provider` already present in `StarkNetNode`:

 * Add a new type parameter `G: Provider` to `StreamService` (maybe need to be `G: Provider + Send + Sync + 'static`).
 * Add a new `provider: Arc<G>` property to the `StreamService`.
 * Add a new argument `provider: Arc<G>` to `new` and assign it to the struct property.
 * Use that provider in this method.